packages/strapi-plugin-upload/admin/src/containers/HomePage/reducer.js

```diff
@@ -40,6 +46,24 @@ const reducer = (state, action) => {
 
       return state.removeIn(['dataToDelete', index]);
     }
+    case 'SET_PARAM': {
+      const { name, value } = action;
+
+      if (name === '_limit') {
+        return state
+          .updateIn(['searchParams', name], () => value)
+          .updateIn(['searchParams', '_start'], () => 0);
+      }
+
+      return state.updateIn(['searchParams', name], () => value);
+    }
+    case 'SET_PARAMS': {
+      const { params } = action;
+
+      return state.update('searchParams', searchParams => {
+        return searchParams.mergeDeep(fromJS(params));
```

Comment:
You should keep the data immutable here. Maybe you can use a `reduce` to merge the new params. Let me know if you are struggling here.